__tests__/registries/npm-registry.js

```diff
@@ -146,15 +164,68 @@ describe('request', () => {
     const {mockRequestManager, mockRegistries, mockReporter} = createMocks();
     const npmRegistry = new NpmRegistry(testCwd, mockRegistries, mockRequestManager, mockReporter);
 
-    const pathname = 'https://registry.npmjs.org/@testScope/yarn.tgz';
+    const url = 'https://registry.npmjs.org/@testScope/yarn.tgz';
 
     npmRegistry.config = {
       _authToken: 'testAuthToken',
     };
-    npmRegistry.request(pathname);
+    npmRegistry.request(url);
 
     const requestParams = mockRequestManager.request.mock.calls[0][0];
 
     expect(requestParams.headers.authorization).toBe('Bearer testAuthToken');
   });
 });
+
+describe('isRequestToRegistry functional test', () => {
+  test('request to registry url matching', () => {
+    const testCwd = '.';
+    const {mockRequestManager, mockRegistries, mockReporter} = createMocks();
+    const npmRegistry = new NpmRegistry(testCwd, mockRegistries, mockRequestManager, mockReporter);
+
+    expect(npmRegistry.isRequestToRegistry('http://foo.bar:80/foo/bar/baz', 'http://foo.bar/foo/')).toBe(true);
+
+    expect(npmRegistry.isRequestToRegistry('http://foo.bar/foo/bar/baz', 'http://foo.bar/foo/')).toBe(true);
+
+    expect(
+      npmRegistry.isRequestToRegistry(
+        'http://foo.bar/foo/00000000-1111-4444-8888-000000000000/baz',
+        'http://foo.bar/foo/',
+      ),
+    ).toBe(true);
+
+    expect(npmRegistry.isRequestToRegistry('https://foo.bar:443/foo/bar/baz', 'https://foo.bar/foo/')).toBe(true);
```

Comment:
You may also wanna look at this library which does all this logic for you: https://yarnpkg.com/en/package/normalize-url